src/components/data_search/DatasetSearchTable.jsx

```diff
@@ -52,7 +57,9 @@ export const DatasetSearchTable = (props) => {
   const [showTranslatedDULModal, setShowTranslatedDULModal] = useState(false);
   const [dataUse, setDataUse] = useState();
   const searchRef = useRef('');
-  const isFiltered = (filter) => filters.indexOf(filter) > -1;
+
+  const isFiltered = (filter, category) => (filters[category]).indexOf(filter) > -1;
+  const numSelectedFilters = (filters) => Object.values(filters).reduce((sum, array) => sum + array.length, 0);
```

Comment:
I'm not super familiar with the lodash reduce function, would you mind explaining what is happening here? Why couldn't we just the length of the array of filters?

Reply:
We can definitely do something like `filters.accessManagement.length + filters.dataUse.length`, but this mechanism avoids having to remember to add new filters to this function. `reduce` is a [Javascript built-in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce), it runs a callback over each element of the array with an accumulator, with 0 as the initial value.

`Object.values(filters)` transforms `{ f1: ['a', 'b', 'c'], f2: ['d', 'e', 'f'] }` into `[ ['a', 'b', 'c'], ['d', 'e', 'f'] ]`
then `reduce` loops over each item in the array (which are subarrays), adds the length of the subarrays to the sum (which starts at zero), and so ultimately returns the total number of selected filters.

Reply:
RE: lodash, that was also my understanding, moving specifically away from the `fp` version of lodash.

Reply:
Thank for explaining that!